File: ClientApp/libraries/core/src/lib/models/page-request.ts


export interface PageRequest {
    pageNumber: number;
    pageSize: number;
    sortParam?: string;
    Desc?: boolean;
    searchParam?: string;
    last?: boolean;
    [key: string]: any;
}

export const REQUESTPARAMS = function get(request: PageRequest): string {
    if (!request) return '';
    let url = `?`;
    for (let [key, value] of Object.entries(request)) {
        if (value === undefined || value === null) continue;
        url += `&${key}=${value}`;
    }
    return url;
}

File: CoreClasses/Pagination/PageRequest.cs
namespace IEduZimAPI.CoreClasses
{
    public class PageRequest
    {
        public int PageNumber { get; set; }
        public int PageSize { get; set; }
        public string SortParam { get; set; }
        public bool Desc { get; set; }
        public bool Active { get; set; }
        public string AdditionalData{ get; set; }
    }
}